src/Core/System/CustomEntity/Schema/SchemaUpdater.php

```diff
@@ -190,6 +190,17 @@ private function addColumns(Schema $schema, Table $table, array $fields): void
                     // get reference name for foreign key building
                     $referenceName = $field['reference'];
 
+                    $ignoreMissingReference = $field['ignoreMissingReference'] ?? false;
+                    if ($ignoreMissingReference) {
+                        /**
+                         * Check if reference table exists
+                         * If not, skip the creation of the many-to-many association
+                         */
+                        if (!$schema->hasTable($referenceName)) {
+                            continue 2;
+                        }
```

Comment:
please throw exception in else case ignore-if-missing =false, is clearer and safer than any silent skipping
 if (!$schema->hasTable($referenceName)) {
    if ($ignoreIfMissing) {
        continue 2; // skip gracefully
    } else {
        CustomEntityException:::"Reference table '$referenceName' does not exist!");
    }
}